### ISSUE TITLE
Email Seems To Be Sent To Wrong Email Address

### ISSUE THREAD

**Issue by PirunSeng** [thredded/thredded#370]
This is such a cool gem.
Thanks to all contributors.

I've been stuck implementing it on my rails app since the email configuration seems not to work well.

I set this in the thredded initializer:
Thredded.email_from = 'myemail@example.com'
Thredded.email_incoming_host = 'localhost:3000'
Thredded.email_reply_to = -> postable { "#{postable.hash_id}@#{Thredded.email_incoming_host}" }

The problems are:
1. Domain contains illegal character
2. From and To : email addresses are the same (myemail@example.com)

Any help would be appreciate.
Thanks,

**Comment by PirunSeng** [comment]
There is no error when I deploy to Staging server with the config host below.
`Thredded.email_incoming_host = "my-hosting-domain.com"`

The email notification is sent to the right email address of the user, but with wrong email address in the email detail.

To be clear, the following is my scenario.
- I login (user1@gmail.com) to the site and post a message. 
- I logout and login again, but with a different account (user2@gmail.com). Then, I replied to the a message above.
- My first login account (user1@gmail.com) receives a notification email and says there is a reply on my message, for example, (This is what I expected.) However, when I click the down arrow icon to see the detail, what I see is that FROM and TO email addresses are the same (myemail@example.com).
What I expect should be:
FROM : myemail@example.com
TO : (user1@gmail.com)

I really need your help.

**Comment by jayroh** [comment]
Incoming mail is no longer supported as a part of the gem, so if you're sending messages in TO your app, it won't work.

If you're talking about sending messages out FROM your app, it would be helpful if you pasted your config related to that in here.

**Comment by PirunSeng** [comment]
Thanks for quick response.
You're right. My purpose is just to send out messages from my app.

Below code are my config:
`Thredded.email_from = "myemail@example.com"`
`Thredded.email_incoming_host = "my-hosting-domain.com"`
`Thredded.email_reply_to = -> postable { "#{postable.hash_id}@#{Thredded.email_incoming_host}" }`

What I see in my gmail inbox:
from: myemail@example.com
to: myemail@example.com

What I want to state is that, the email should be sent to user1@gmail.com.
It should be like the following:
from: myemail@example.com
to: user1@gmail.com

**Comment by glebm** [comment]
The same email is sent to multiple users who a listed in the BCC field. The to field is not populate in order to avoid exposing the same email to multiple users. See: https://thredded.org/emails/en/thredded__post_mailer_preview-post_notification

**Comment by PirunSeng** [comment]
Oh, I see.
However, I don't see the BCC field in my email.
Is there other configuration I have to set or override?

**Comment by PirunSeng** [comment]
Alright, I can override the 'TO' email address, just in my opinion, I prefer to see the email is sent to my email address, so that I can realize it is sent to the right person.

I override thredded/app/mailers/thredded/post_mailer.rb

`# frozen_string_literal: true
require_dependency 'thredded/topic_email_view'
module Thredded
  class PostMailer < Thredded::BaseMailer
    def post_notification(post_id, emails)
      @post                = find_record Post, post_id
      email_details        = TopicEmailView.new(@post.postable)
      headers['X-SMTPAPI'] = email_details.smtp_api_tag('post_notification')

      mail from:     email_details.no_reply,
           to:       email_details.no_reply,
           bcc:      emails,
           reply_to: email_details.reply_to,
           subject:  email_details.subject
    end
  end
end`

To

`# frozen_string_literal: true
require_dependency 'thredded/topic_email_view'
module Thredded
  class PostMailer < Thredded::BaseMailer
    def post_notification(post_id, emails)
      @post                = find_record Post, post_id
      email_details        = TopicEmailView.new(@post.postable)
      headers['X-SMTPAPI'] = email_details.smtp_api_tag('post_notification')

      mail from:     email_details.no_reply,
           to:       emails,
           bcc:      emails,
           reply_to: email_details.reply_to,
           subject:  email_details.subject
    end
  end
end`

In summary, I just change:
`to: email_details.no_reply` to `to: emails`

**Comment by PirunSeng** [comment]
I can close this issue for now since I can override the TO field.
However, I just want to leave a note that BCC field is still not shown in the email.
Thank guys for your response that can help me to think about overriding the mailer.
:+1:

**Comment by ksaleem** [comment]
thanks done in same way
